apps/console/src/components/pages/protected/onboarding/step-2.tsx

```diff
@@ -0,0 +1,57 @@
+'use client'
+import { Input } from '@repo/ui/input'
+import { Label } from '@repo/ui/label'
+import { Select, SelectContent, SelectItem, SelectTrigger, SelectValue } from '@repo/ui/select'
+import React from 'react'
+import { useFormContext } from 'react-hook-form'
+import { z, infer as zInfer } from 'zod'
+
+export const step2Schema = z.object({
+  userDetails: z
+    .object({
+      role: z.string().optional(),
+      department: z.string().optional(), // ✅ Made department optional
+    })
+    .optional(),
+})
+
+type Step2Values = zInfer<typeof step2Schema>
+
+export default function Step2() {
+  const {
+    register,
+    formState: { errors },
+    setValue,
+    watch,
+  } = useFormContext<Step2Values>()
+
+  return (
+    <div className="space-y-6">
+      <h2 className="text-xl font-semibold">User Info</h2>
+
+      {/* User Role */}
+      <div className="space-y-2">
+        <Label htmlFor="role">Role*</Label>
```

Comment:
Role is not required
```suggestion
        <Label htmlFor="role">Role</Label>
```